django_statsd/middleware.py

```diff
@@ -5,24 +5,28 @@
 from django.http import Http404
 
 from django_statsd.clients import statsd
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:
+    MiddlewareMixin = object
 
 
-class GraphiteMiddleware(object):
+class GraphiteMiddleware(MiddlewareMixin):
 
     def process_response(self, request, response):
         statsd.incr('response.%s' % response.status_code)
-        if hasattr(request, 'user') and request.user.is_authenticated():
+        if hasattr(request, 'user') and request.user and request.user.is_authenticated():
```

Comment:
Any test case or at least use case that would explain this part ?

Reply:
I'm running into an issue with this now. I believe its because django's test Client has the user attribute by default but it is set to `None` in the case where the Client doesn't have a logged in user. I would write this as 
```
if getattr(request, 'user', None) and request.user.is_authenticated():
```

Reply:
pull request to fix just the test client issue: https://github.com/django-statsd/django-statsd/pull/108